tests/test_optics.py

```diff
@@ -53,3 +54,77 @@ def test_sunrise_sunset_fail():
 
     with pytest.raises(ValueError):
         sunrise, sunset = gt.optics.sunset_sunrise(time, lat, lon)
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_backscatter_dark_count(percentile):
+    from glidertools.optics import backscatter_dark_count
+
+    # create some synthetic data
+    bbp = np.array([0.002, 0.0006, 0.0005, 0.0005, 0.0005])
+    depth = np.array([50, 150, 210, 310, 350])
+    # select only depths between 200 and 400
+    mask = (depth > 200) & (depth < 400)
+    # expected output
+    expected_bbp_dark = bbp - np.nanpercentile(bbp[mask], percentile)
+    bbp_dark = backscatter_dark_count(bbp, depth, percentile)
+    np.testing.assert_allclose(expected_bbp_dark, bbp_dark)
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_backscatter_dark_count_negative(percentile):
+    from glidertools.optics import backscatter_dark_count
+
+    # create some synthetic data
+    bbp = np.array(
+        [0.002, 0.0006, 0.005, 0.005, 0.0004]
+    )  # this will result in negative values that should be zeroed out
+    depth = np.array([50, 150, 210, 310, 350])
+    bbp_dark = backscatter_dark_count(bbp, depth, percentile)
+    # in this case we just want to check if none of the values is negative!
+    assert np.all(bbp_dark >= 0)
+
```

Comment:
Sorry for not noting this earlier. I just saw that there is a warning in the `backscatter_dark_count` function [here](https://github.com/GliderToolsCommunity/GliderTools/blob/5123801fdc964180436ee4f716e3e27ea6727042/glidertools/optics.py#L150-L155). Could you add another test that triggers this warning and assert that it is raised correctly? 

This could look something like this:
```python
def test_backscatter_dark_count_warning():
    # create some data that would trigger the warning (I think it should be just really shallow depth?)
   with pytest.warns(UserWarning): # this line will fail if the command below does not actually raise a warning!
        backscatter_dark_count(bbp, depth, percentile)
```

for more information on how to assert warnings with pytest see the [documentation](https://docs.pytest.org/en/doc-warnings/recwarn.html)